docs/DeveloperGuide.md

```diff
@@ -1,34 +1,837 @@
+episodes for each season respectively,Duration of an episode)
+
+
+**Step 4**
+
+* A new `Show` instance is created with the 4 parameters created in step 3 using the default constructor for show.
+
+**Step 5**
+
+* The `Show` is added to the `ShowList`.
+
+
+**Step 6** 
+
+* The changes will be reflected to the user. Concurrently, the changes will be saved into the showList.txt file.
+
+### EditCommand
+
+The `edit` command allows the user to change the details of each show that they are watching after they have added the
+show. It is self-contained, it includes its own parser and methods which prompts the user to change any parameter they 
+wish, after the user enters `done`, the `edit` command replaces the old entry with the updated one.
+
+Given below is an example of usage scenario of how the edit command behaves at each step and a sequence diagram to
+illustrate the steps in a visual form.
+
+> **[NOTE]** 
+>
+> `NullPointerException` will be thrown when show entered by user is not found in the showlist.
+
+<img src = "images/EditCommand.png" width = "700">
+ 
+ <br> 
+*Figure 1: Sequence Diagram for Edit Command*
+
+<img src = "images/edit_step_1.PNG" width = "400">
+   
+*Figure 2: Program FLow for Edit Command*
+
+**Step 1**
+
+* The user types in `edit friends` , where the show `friends` already exists in the showlist.
+
+**Step 2**
+
+* The `processCommand` method is called. The `processCommand` method will retrieve the existing show object from the showlist,
+and make a copy of it.
+* Then the system will prompt the user to edit the name,season,episodes or the 
+duration (of an episode) respectively. 
+
+**Step 3**
+
+* The process command parses each line of the user input.
+* The `EditCommand` class then calls the corresponding method, e.g. `editDuration()` to make the corresponding changes to the copy.
+
+**Step 4**
+
+* The user inputs `done`, and the copy of the `Show` object is inserted into the `ShowList`, replacing the old object.
+
+
+
+### DeleteCommand
+  
+The `delete` command takes in 1 parameter, the show to be deleted. Following that, the command proceeds to delete the
+`Show` from the `ShowList`. The `delete` command is invoked by the `InputParser` Method `parseDeleteCommand`.
+
+
+Given below is an example usage scenario and how the `DeleteCommand` Class behaves at each step.
+
+**Step 1**
+
+* The user types in `delete friends` , assuming that friends has been added by the user beforehand.
+The `parseInput` method in `InputParser` class is called to parse the command.
+
+> **[NOTE]** 
+>
+> Customised `NullPointerException` will be thrown when show entered by user is not found in the show list.
+
+**Step 2**
+
+* A new instance of the `DeleteCommand` class is called and the command is returned to the main program. 
+The `delete` method in `DeleteCommand` class is called.
+
+**Step 3**
+
+* The `delete` method retrieves the `Show` to be deleted from the `ShowList`.
+
+**Step 4**
+
```

Comment:
Perhaps can consider adding sequence diagrams for all the commands to integrate it with the steps provided